lib/src/map/map.dart

```diff
@@ -338,17 +341,17 @@ class MapState {
       throw Exception('Bounds are not valid.');
     }
     var target = getBoundsCenterZoom(bounds, options);
-    move(target.center, target.zoom);
+    move(target.center, target.zoom, source: MapEventSource.fitBounds);
   }
 
-  LatLng getCenter() {
+  LatLng? getCenter() {
     if (_lastCenter != null) {
       return _lastCenter;
     }
     return layerPointToLatLng(_centerLayerPoint);
   }
 
-  LatLngBounds getBounds() {
+  LatLngBounds? getBounds() {
```

Comment:
Is there a reason getBounds() returns a `LatLngBounds?` instead of a `LatLngBounds` object? As far as I can see, null cannot be returned here because `_calculateBounds()` already returns a `LatLngBounds` object.

Reply:
Agreed, don't know why I out it there. Same with the `bounds` property but I wonder why do we have a get function and a property?

Reply:
Right.  The same goes for line 164:
```dart
LatLng get center => getCenter() ?? options.center;
```
getCenter() can also never return null. 

It seems that the get properties are generally used and the getBounds() and getCenter() functions are more internal functions. So it's really worth considering if they really make sense.

Reply:
> Agreed, don't know why I out it there. Same with the `bounds` property but I wonder why do we have a get function and a property?

Changing it from function to property would be a breaking. I guess this is a good time to to some breaking improvements as any.

Reply:
@josxha The dart analyzer does not allow `LatLongBounds` as return-type because `_lastBounds` is defined as nullable with `LatLngBounds? _lastBounds`. I don't know if this is an error on the dart-side since the if-check above clearly removes the possibility of returning null. I guess there is something else going on here (like me not understanding every aspect of sound null-safety and thus how to handle cases like these) 

<img width="849" alt="Screenshot 2021-04-25 at 14 18 49" src="https://user-images.githubusercontent.com/866528/115993150-ce21c680-a5d1-11eb-870f-6fecf301aae5.png">

Reply:
@kengu regarding the null check, despite the not null check you need to "Cast away the null". So the

Reply:
> @kengu regarding the null check, despite the not null check you need to "Cast away the null". So the
> `return _lastBounds;` should be `return _lastBounds!;`
> 
> also see: https://stackoverflow.com/questions/63253015/what-does-the-exclamation-mark-mean-before-a-function-call
> the official null safety documentation does suggest it's not necessary, however I've encountered similar issues and resolved them similarly.

Check - will look into it

Reply:
Seems reasonable enough to use `!`, event though static safety is lost. See [null assertion operator](https://dart.dev/null-safety/understanding-null-safety#null-assertion-operator)

> Of course, like any cast, using ! comes with a loss of static safety. The cast must be checked at runtime to preserve soundness and it may fail and throw an exception. But you have control over where these casts are inserted, and 
 you can always see them by looking through your code.

I'll change `getCenter()` and `getBounds()`it soon

Reply:
Changes pushed now